src/ipa_pc/mod.rs

```diff
@@ -1151,30 +781,136 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
             coeffs = coeffs_l;
             z = z_l;
 
-            key_proj = key_proj_l;
-            temp = G::Projective::batch_normalization_into_affine(key_proj.to_vec());
-            comm_key = &temp;
+            key_proj = key_proj_l;
+            temp = G::Projective::batch_normalization_into_affine(key_proj.to_vec());
+            comm_key = &temp;
+
+            n /= 2;
+        }
+
+        end_timer!(proof_time);
+
+        Ok(Proof {
+            l_vec,
+            r_vec,
+            final_comm_key: comm_key[0],
+            c: coeffs[0],
+            hiding_comm,
+            rand,
+        })
+    }
+
+    /// Single point multi poly open, allowing the random oracle to be passed from
+    /// 'outside' to the function.
+    /// CAUTION: This is a low-level function which assumes that the statement of the
+    /// opening proof (i.e. commitments, query point, and evaluations) is already bound
+    /// to the internal state of the Fiat-Shamir rng.
+    fn single_point_multi_poly_open<'a>(
+        ck: &Self::CommitterKey,
+        labeled_polynomials: impl IntoIterator<Item = &'a LabeledPolynomial<G::ScalarField>>,
+        commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
+        point: G::ScalarField,
+        // This implementation assumes that commitments, query point and evaluations are already absorbed by the Fiat Shamir rng
+        fs_rng: &mut Self::RandomOracle,
+        rands: impl IntoIterator<Item = &'a LabeledRandomness<Self::Randomness>>,
+        rng: Option<&mut dyn RngCore>,
+    ) -> Result<Self::Proof, Self::Error>
+        where
+            Self::Commitment: 'a,
+            Self::Randomness: 'a,
```

Comment:
As commented in the Polycommit trait: we can do a default implementation of single_point_multi_poly_open.

Reply:
Done